remote_exec/remote_exec.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+option go_package = "teleport_exec/remote_exec";
+package remote_exec;
+
+//-----------------------------------------------------------------------------
+message StartCommandRequest { string command = 1; }
+
+//-----------------------------------------------------------------------------
+message CommandStatusRequest { string command_id = 1; }
+
+message CommandStateResponse {
+  string command_id = 1;
+  string command = 2;
+  bool running = 3;
+  optional int32 result_code = 4;
+}
```

Comment:
How will you distinguish stopped by signal or exited on own?

Reply:
I was thinking it'd work just like it does in Docker - you get an exit code = 128 + signal number and you know the process was killed (e.g. 137 for sigkill)

Reply:
You can do that, but is it reliable? I could exit with `137` but not really be killed by SIGKILL right?

Reply:
Looking at the [go stdlib](https://github.com/golang/go/blob/master/src/syscall/syscall_linux.go#L320-L345), they simply use a `0x7f` mask on the exit code and rely on it as the way to detect if a process exited or was killed by a signal. 

Anyways, I can add an `optional bool exited` field and populate it with the results from the `WaitStatus` struct if you think it is needed.

Reply:
Sure why not. It's a small lift at the application layer that gives the user a more reliable answer.

Reply:
Added the field to the protocol.